bottlecap/src/metrics/aggregator.rs

```diff
@@ -239,32 +246,78 @@ impl<const CONTEXTS: usize> Aggregator<CONTEXTS> {
             .try_into()
             .unwrap_or_default();
         for entry in &self.map {
-            if entry.kind != metric::Type::Distribution {
+            let Some(sketch) = self.build_sketch(now, entry) else {
                 continue;
-            }
-            let sketch = entry.metric_value.get_sketch();
-            let mut dogsketch = Dogsketch::default();
-            sketch.merge_to_dogsketch(&mut dogsketch);
-            // TODO(Astuyve) allow users to specify timestamp
-            dogsketch.set_ts(now);
-            let mut sketch = Sketch::default();
-            sketch.set_dogsketches(vec![dogsketch]);
-            let name = entry.name.to_string();
-            sketch.set_metric(name.clone().into());
-            let mut base_tag_vec = self.tags_provider.get_tags_vec();
-            let mut tags = tags_string_to_vector(entry.tags);
-            base_tag_vec.append(&mut tags); // TODO split on comma
-            sketch.set_tags(
-                base_tag_vec
-                    .into_iter()
-                    .map(std::convert::Into::into)
-                    .collect(),
-            );
+            };
             sketch_payload.sketches.push(sketch);
         }
         sketch_payload
     }
 
+    #[must_use]
+    pub fn distributions_to_protobuf_serialized(&self) -> Vec<u8> {
+        let mut buffer: Vec<u8> = Vec::with_capacity(self.max_content_size_bytes);
+        let mut count_entries = 0;
+        let now = time::SystemTime::now()
+            .duration_since(time::UNIX_EPOCH)
+            .expect("unable to poll clock, unrecoverable")
```

Comment:
Is there something we can do about this expect? Can we bubble it up?

Reply:
This was existing existing code I moved. I think it's worth to address all the expect/panics in a specific PR